irohad/consensus/yac/storage/yac_vote_storage.hpp

```diff
@@ -128,13 +107,43 @@ namespace iroha {
          */
         void nextProcessingState(const Round &round);
 
+        /// Get the highest round with proposal in kSentProcessed state, if any,
+        /// otherwize boost::none.
+        boost::optional<Round> getLastCompletedRound() const;
+
+        /// Get the outcome of given round, if any, otherwize boost::none.
+        boost::optional<Answer> getRoundOutcome(const Round &round) const;
+
        private:
+        using ProposalStoragePtr = std::shared_ptr<YacProposalStorage>;
+        using OptProposalStoragePtr = boost::optional<ProposalStoragePtr>;
+
+        // --------| private api |--------
+
+        /**
+         * Retrieve iterator for storage with specified key
```

Comment:
Are you really retrieving an iterator? As I see, optional with the storage is returned here